src/bp/ui-studio/src/web/views/OneFlow/diagram/VariableTypesForm/PatternForm.tsx

```diff
@@ -0,0 +1,150 @@
+import { Tab, Tabs } from '@blueprintjs/core'
+import axios from 'axios'
+import sdk from 'botpress/sdk'
+import { Contents, lang, MoreOptions, MoreOptionsItems, RightSidebar } from 'botpress/shared'
+import _ from 'lodash'
+import React, { FC, Fragment, useEffect, useRef, useState } from 'react'
+
+import style from '../PromptForm/style.scss'
+
+import { getEntityId } from '.'
+
+interface Props {
+  customKey: string
+  contentLang: string
+  formData: sdk.NLU.EntityDefinition
+  close: () => void
+  deleteEntity: (entityId: string) => void
+  updateEntity: (originalId: string, entity: sdk.NLU.EntityDefinition) => void
+  updateFormItem: (entity) => void
+}
+
+const preparePattern = (pattern: string, matchCase?: boolean) => {
+  try {
+    let p = pattern || ''
+    if (!p.startsWith('^')) {
+      p = `^${p}`
+    }
+    if (!p.endsWith('$')) {
```

Comment:
`|| endsWith('\\$')` (same for ^)

Reply:
Nop, this would add them even if they are already there